src/utils.py

```diff
@@ -1860,11 +1861,62 @@ def sample_subsets(universe: Sequence[_T], num_samples: int, min_set_size: int,
 def create_ground_atom_dataset(
         trajectories: Sequence[LowLevelTrajectory],
         predicates: Set[Predicate]) -> List[GroundAtomTrajectory]:
-    """Apply all predicates to all trajectories in the dataset."""
-    ground_atom_dataset = []
-    for traj in trajectories:
-        atoms = [abstract(s, predicates) for s in traj.states]
-        ground_atom_dataset.append((traj, atoms))
+    """Generate or load a GroundAtomTrajectory by applying all predicates to
+    all trajectories in the dataset.
+
+    If generating a new GroundAtomTrajectory, then save to file for
+    later reuse.
+    """
+    # Setup the dataset filename for saving/loading GroundAtoms.
+    regex = r"(\d+)"
+    if CFG.env == "behavior":  # pragma: no cover
+        dataset_fname_template = (
+            f"{CFG.env}__{CFG.behavior_scene_name}__{CFG.behavior_task_name}" +
+            f"__{CFG.offline_data_method}__{CFG.demonstrator}__"
+            f"{regex}__{CFG.included_options}__{CFG.seed}__ground_atoms.data")
+    else:
+        dataset_fname_template = (
+            f"{CFG.env}__{CFG.offline_data_method}__{CFG.demonstrator}__"
+            f"{regex}__{CFG.included_options}__{CFG.seed}__ground_atoms.data")
+    dataset_fname = os.path.join(
+        CFG.data_dir,
+        dataset_fname_template.replace(regex, str(CFG.num_train_tasks)))
+    # If CFG.load_atoms is set, then try to load the GroundAtomTrajectory
+    # directly from a saved file.
+    if CFG.load_atoms:
+        os.makedirs(CFG.data_dir, exist_ok=True)
+        # Check that the dataset file was previously saved.
+        if os.path.exists(dataset_fname):
+            # Load the ground atoms dataset.
+            with open(dataset_fname, "rb") as f:
+                ground_atom_dataset_trjectories = pkl.load(f)
+            logging.info("\n\nLOADED GROUND ATOM DATASET")
+            ground_atom_dataset = []
+            for i, traj in enumerate(trajectories):
+                ground_atom_seq = ground_atom_dataset_trjectories[i]
+                ground_atom_dataset.append(
+                    (traj, [set(atoms) for atoms in ground_atom_seq]))
+        else:
+            raise ValueError(f"Cannot load ground atoms: {dataset_fname}")
+    else:
+        ground_atom_dataset = []
+        for traj in trajectories:
+            atoms = [abstract(s, predicates) for s in traj.states]
+            ground_atom_dataset.append((traj, atoms))
+        # Save ground atoms dataset to file.
+        ground_atom_dataset_to_pkl = []
+        for _, traj in enumerate(ground_atom_dataset):
+            trajectory = []
+            for i, ground_atom_seq in enumerate(traj[1]):
+                trajectory.append({
+                    GroundAtom(
+                        Predicate(atom.predicate.name, atom.predicate.types,
+                                  lambda s, o: None), atom.entities)
```

Comment:
major: you're stripping out the classifiers, presumably for behavior related reasons, but you're not putting them back in during loading. that seems risky to me -- can you fix that, and add a test to make sure that the loaded atoms have the right classifiers attached to them?

minor: add a comment explaining why we need to do this stripping

minor: I'm surprised mypy let's you get away with `None` here, shouldn't it be a bool?

Reply:
yup - i had to fix the `bool` thing. It didn't strike me that this stripping was always being done and that's potentially dangerous - will definitely make sure it's (1) only for BEHAVIOR and (2) will test classifiers after loading are correct!